modules/cloud-credential-operator.adoc

```diff
@@ -8,8 +8,275 @@
 [discrete]
 == Purpose
 
-The Cloud Credential Operator manages cloud provider credentials as Kubernetes
-custom resource definitions (CRDs).
+The Cloud Credential Operator (CCO) manages cloud provider credentials as
+Kubernetes custom resource definitions (CRDs). The CCO syncs on
+`credentialsRequest` custom resources (CRs) to allow {product-title} components
+to request cloud provider credentials with the specific permissions that are
+required for the cluster to run.
+
+By setting different values for the `credentialsMode` parameter in the
+`install-config.yaml` file, the CCO can be configured to operate in several
+different modes. If no mode is specified, or the `credentialsMode` parameter is
+set to an empty string (`""`), the CCO operates in its default mode.
+
+[discrete]
+=== Default behavior
+For platforms where multiple modes are supported (AWS, Azure, and GCP), when the CCO operates in its default mode, it checks the provided credentials
+dynamically to determine for which mode they are sufficient to process `credentialsRequest` CRs.
+
+By default, the CCO determines whether the credentials are sufficient for mint
+mode, which is the preferred mode of operation, and uses those credentials to
+create appropriate credentials for components in the cluster. If the credentials
+are not sufficient for mint mode, it determines whether they are sufficient for
+passthrough mode. If the credentials are not sufficient for passthrough mode,
+the CCO cannot adequately process `credentialsRequest` CRs.
+
+[NOTE]
+====
+The CCO cannot verify whether Azure credentials are sufficient for passthrough mode. If Azure credentials are insufficient for mint mode, the CCO operates with the assumption that the credentials are sufficient for passthrough mode.
+====
+
+If the provided credentials are determined to be insufficient during
+installation, the installation fails. For AWS, the installer fails early in the
+process and indicates which required permissions are missing. Other providers
+might not provide specific information about the cause of the error until errors are encountered.
+
+If the credentials are changed after a successful installation and the CCO
+determines that the new credentials are insufficient, the CCO puts conditions on
+any new `credentialsRequest` CRs to indicate that it cannot process them because
+of the insufficient credentials.
+
+To resolve insufficient credentials issues, provide a credential with sufficient
+permissions. If an error occurred during installation, try installing again. For
+issues with new `credentialsRequest` CRs, wait for the CCO to try to process the
+CR again. As an alternative, you can manually create IAM for
+xref:../installing/installing_aws/manually-creating-iam.adoc[AWS],
```

Comment:
@jeana-redhat FYI we can't include `xef` in modules.

Reply:
Same comment about XREF